[tradfri] NPE if gateway is unreachable

In case the IKEA gateway is unreachable the binding runs into an NPE:

```
2021-01-01 15:31:32.243 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tradfri.internal.handler.TradfriControllerHandler@6f16b0c4': null
java.lang.NullPointerException: null
        at org.eclipse.californium.core.CoapClient.setEndpoint(CoapClient.java:354) ~[?:?]
        at org.openhab.binding.tradfri.internal.handler.TradfriThingHandler.initialize(TradfriThingHandler.java:73) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor58.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:154) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
```

The reason being that `handler.getEndpoint()` returns `null` here:

https://github.com/openhab/openhab-addons/blob/d3ae64140f958f5e945d08d4122e590549405ef3/bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriThingHandler.java#L67-L73

I noticed that the gateway `handler` (bridge) does not have an endpoint if it is `OFFLINE`.

All happened during the initialization, so after the bridge goes `OFFLINE` the initialization of the handlers is triggered and then runs into the NPE.

```
2021-01-01 15:31:32.228 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tradfri:gateway:ikeagw' changed from INITIALIZING to OFFLINE (COMMUNICATION_ERROR): No response from gateway. Might be due to an invalid security code.
2021-01-01 15:31:32.233 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tradfri:0830:ikeagw:ikeaRemote' changed from UNINITIALIZED (BRIDGE_UNINITIALIZED) to INITIALIZING
2021-01-01 15:31:32.235 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tradfri:0210:ikeagw:ikeaColor' changed from UNINITIALIZED (BRIDGE_UNINITIALIZED) to INITIALIZING
2021-01-01 15:31:32.261 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tradfri:0830:ikeagw:ikeaRemote' changed from INITIALIZING to UNINITIALIZED (HANDLER_INITIALIZING_ERROR)
2021-01-01 15:31:32.261 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tradfri:0210:ikeagw:ikeaColor' changed from INITIALIZING to UNINITIALIZED (HANDLER_INITIALIZING_ERROR)
```

I will provide a PR that sets the `coapEndpoint` only if the bridge is `ONLINE`.

--- a/bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriThingHandler.java
+++ b/bundles/org.openhab.binding.tradfri/src/main/java/org/openhab/binding/tradfri/internal/handler/TradfriThingHandler.java
@@ -66,20 +66,21 @@ public abstract class TradfriThingHandler extends BaseThingHandler implements Co
         this.id = getConfigAs(TradfriDeviceConfig.class).id;
         TradfriGatewayHandler handler = (TradfriGatewayHandler) tradfriGateway.getHandler();
 
-        String uriString = handler.getGatewayURI() + "/" + id;
-        try {
-            URI uri = new URI(uriString);
-            coapClient = new TradfriCoapClient(uri);
-            coapClient.setEndpoint(handler.getEndpoint());
-        } catch (URISyntaxException e) {
-            logger.debug("Illegal device URI `{}`: {}", uriString, e.getMessage());
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
-            return;
-        }
         active = true;
         updateStatus(ThingStatus.UNKNOWN);
         switch (tradfriGateway.getStatus()) {
             case ONLINE:
+                String uriString = handler.getGatewayURI() + "/" + id;
+                try {
+                    URI uri = new URI(uriString);
+                    coapClient = new TradfriCoapClient(uri);
+                    coapClient.setEndpoint(handler.getEndpoint());
+                } catch (URISyntaxException e) {
+                    logger.debug("Illegal device URI `{}`: {}", uriString, e.getMessage());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                    return;
+                }
+
                 scheduler.schedule(() -> {
                     observeRelation = coapClient.startObserve(this);
                 }, 3, TimeUnit.SECONDS);

